frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -99,6 +103,7 @@ export function InstantSearchProvider({
                }),
             }}
          >
+            <RefreshSearchResults apiKey={algoliaApiKey} />
```

Comment:
> When would InstantSearch unmount / remount?

We explicitly do this because of *another* bug in Algolia. sEe the fix here: https://github.com/iFixit/react-commerce/pull/351

> this component will think that isFirstRender is true, yes?

No, cause the ref is created on line 41 in the `AppProviders` context which doesn't re-mount and thus is stable. See line 128.

I experimented with a bunch of techniques till I got only *one* algolia search happening when we swap the key.

Reply:
Gotcha thanks!